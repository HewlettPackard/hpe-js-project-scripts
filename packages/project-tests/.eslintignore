node_modules/
coverage/
dist/
build/
prod/
out/
.next/
.cache/
public/
yarn.lock
package-lock.json