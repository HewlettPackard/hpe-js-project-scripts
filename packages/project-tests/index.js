const reactTestRenderer = require('react-test-renderer');
const jestNextConfig = require('./jest.next-config');

module.exports = { jestNextConfig, reactTestRenderer };
