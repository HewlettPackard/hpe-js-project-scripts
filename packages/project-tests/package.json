{
  "name": "@hpe/project-tests",
  "version": "1.0.0",
  "description": "HPE JS project tests",
  "main": "index.js",
  "repository": {
    "type": "git",
    "url": " https://github.com/HewlettPackard/hpe-js-project-scripts.git"
  },
  "scripts": {
    "build": "rm -rf dist && webpack --mode production"
  },
  "author": "Alex Mejias",
  "license": "MIT",
  "dependencies": {},
  "devDependencies": {
    "babel-loader": "^8.0.6",
    "copy-webpack-plugin": "^5.0.4",
    "webpack": "^4.38.0",
    "webpack-cli": "^3.3.6"
  }
}
