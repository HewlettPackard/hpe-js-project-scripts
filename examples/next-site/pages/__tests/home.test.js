import React from 'react';
import { reactTestRenderer } from '@hpe/project-tests';
import Home from '../index';
import PageTwo from '../pageTwo';

test('Home renders', () => {
  const component = reactTestRenderer.create(
    <Home data={{ name: 'test', main: { temp: 7000 } }} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Page Two renders', () => {
  const component = reactTestRenderer.create(
    <PageTwo data={{ name: 'test', main: { temp: 7000 } }} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
