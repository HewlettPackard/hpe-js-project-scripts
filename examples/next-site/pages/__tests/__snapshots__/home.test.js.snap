// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 kwUvnO"
>
  <h1
    className="StyledHeading-sc-1rdh4aw-0 dpOUKP"
    size="xlarge"
  >
    Home Page
  </h1>
  <h1
    className="StyledHeading-sc-1rdh4aw-0 rfVJR"
  >
    test
     is 
    7000
     degrees.
  </h1>
  <a
    href="/pageTwo"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Next Page
  </a>
</div>
`;

exports[`Page Two renders 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 kwUvnO"
>
  <h1
    className="StyledHeading-sc-1rdh4aw-0 dpOUKP"
    size="xlarge"
  >
    Next Page
  </h1>
  <h1
    className="StyledHeading-sc-1rdh4aw-0 rfVJR"
  >
    test
     is 
    7000
     degrees.
  </h1>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Home Page
  </a>
</div>
`;
